app/uk/gov/hmrc/merchandiseinbaggagefrontend/controllers/DeclarationJourneyController.scala

```diff
@@ -34,6 +37,12 @@ trait IndexedDeclarationJourneyController extends FrontendBaseController {
   implicit def messages(implicit request: Request[_]): Messages = controllerComponents.messagesApi.preferred(request)
 
   def onPageLoad(idx: Int): Action[AnyContent]
+
+  def withGoodsCategory(goodsEntry: GoodsEntry)(f: String => Future[Result]): Future[Result] =
```

Comment:
this is definitely an improvement 

but for the win (maybe in another pr for another ticket) you could have something like a 

```

def goodsCategoryAction(idx: Int): ActionBuilder[DeclarationGoodsRequest, AnyContent] = goodsAction(idx) andThen goodsCategoryRefiner

goodsCategoryRefiner: ActionRefiner[DeclarationGoodsRequest, DeclarationGoodsCategoryRequest] = ...
```

which would remove more repetition from the controllers and give you everything in one bite.